packages/@aws-cdk/aws-apigatewayv2-integrations/test/http/aws.test.ts

```diff
@@ -0,0 +1,86 @@
+import '@aws-cdk/assert-internal/jest';
+import { HttpApi, HttpRoute, HttpRouteKey } from '@aws-cdk/aws-apigatewayv2';
+import { StateMachine, Chain, Pass, StateMachineType } from '@aws-cdk/aws-stepfunctions';
+import { Stack, Duration } from '@aws-cdk/core';
+import { StepFunctionsStartExecutionIntegration, StepFunctionsStartSyncExecutionIntegration } from '../../lib';
+
+describe('AwsServiceIntegration', () => {
+  test('StepFunctions-StartExecution', () => {
+    const stack = new Stack();
+    const api = new HttpApi(stack, 'HttpApi');
+
+    new HttpRoute(stack, 'StepFunctionsStartExeRoute', {
+      httpApi: api,
+      integration: new StepFunctionsStartExecutionIntegration(stack, {
+        stateMachine: stateMachine(stack),
+        name: 'MyExe',
+        input: '$request.body',
+        timeout: Duration.seconds(10),
+        description: 'Start execution of state machine',
+      }),
+      routeKey: HttpRouteKey.with('/start'),
+    });
+
+    expect(stack).toHaveResource('AWS::ApiGatewayV2::Integration', {
+      IntegrationType: 'AWS_PROXY',
+      CredentialsArn: {
+        'Fn::GetAtt': [
+          'StepFunctionsStartExeRouteApiGatewayIntegrationRoleE88875A9',
+          'Arn',
+        ],
+      },
+      IntegrationSubtype: 'StepFunctions-StartExecution',
+      PayloadFormatVersion: '1.0',
+      RequestParameters: {
+        StateMachineArn: {
+          Ref: 'MyStateMachine6C968CA5',
```

Comment:
I wonder if this should be the state machine arn instead of the resource. Did it work if you try to deploy?

Reply:
nope it does not work. it needs the resource.